### DISCUSSION TITLE
No VS Code Remote Development (SSH) extension and no GitHub CoPilot?

### DISCUSSION THREAD

**Discussion by progettoautomazione** [coder/code-server#6582] [Q&A]
Hi I was all excited to use Coder on my brower, I installed it on my ubuntu and in the Extensions Search panel I cannot find neither VS Code Remote Development (SSH) nor CoPilot extension. I tried installing by VISX for Remote Development and I get the message that these axtensions are not suppported. Is this a known limit? Any workaround? Thanks

**Comment by code-asher** [comment]
code-server uses https://open-vsx.org/ as the extension marketplace and neither of those extensions are hosted there because they are closed-source and Microsoft has opted not to distribute them there.  Technically you could make code-server use Microsoft's marketplace but this is against the terms of service (it is only allowed to be used with Microsoft-distributed VS Code).

Supposedly the copilot extension does work if installed manually but the SSH extension I believe simply does not work at all.  I am not sure if the copilot license allows running in non-Microsoft versions of VS Code, though.  My guess would be no.
